File: Spring Boot Implementations/Personal-Budget-Tracker-App/src/main/java/in/edu/eec/easwari/Personal/Budget/Tracker/App/service/BudgetService.java
package in.edu.eec.easwari.Personal.Budget.Tracker.App.service;

import java.util.List;

import in.edu.eec.easwari.Personal.Budget.Tracker.App.dto.request.BudgetRequestDTO;
import in.edu.eec.easwari.Personal.Budget.Tracker.App.dto.response.BudgetResponseDTO;

public interface BudgetService {

    BudgetResponseDTO setBudget(BudgetRequestDTO request);

    List<BudgetResponseDTO> getMonthlyBudgets(Long userId, Integer month, Integer year);
}

File: ReactJS Implementations/personal-budget-tracker/src/services/budgetService.js
import api from "../api/axiosConfig";

export const addBudget = (data) => 
  api.post("/budget", data);

export const getMonthlyBudgets = (userId, month, year) => 
  api.get(`/budget/${userId}/${month}/${year}`);


export const getBudgetUsage = (userId) => 
  api.get(`/budget/usage/${userId}`);
